ironic/resource_ironic_node_v1.go

```diff
@@ -585,3 +607,83 @@ func changePowerState(client *gophercloud.ServiceClient, d *schema.ResourceData,
 
 	return nil
 }
+
+// setRAIDConfig calls ironic's API to send request to change a Node's RAID config.
+func setRAIDConfig(client *gophercloud.ServiceClient, d *schema.ResourceData) (err error) {
+	var logicalDisks []nodes.LogicalDisk
+	var targetRAID *metal3v1alpha1.RAIDConfig
+
+	raidConfig := d.Get("raid_config").(string)
+	if raidConfig == "" {
+		return nil
+	}
+
+	err = json.Unmarshal([]byte(raidConfig), &targetRAID)
+	if err != nil {
+		return
+	}
+
+	err = ironic.CheckRAIDInterface(d.Get("raid_interface").(string), targetRAID)
+	if err != nil {
+		return
+	}
+
+	// Build target for RAID configuration steps
+	logicalDisks, err = ironic.BuildTargetRAIDCfg(targetRAID)
+	if len(logicalDisks) == 0 || err != nil {
```

Comment:
nit: I don't think the `len` check is necessary: if there is an error, we should always return

Reply:
> if there is an error, we should always return

Yes, it is.

> I don't think the `len` check is necessary

If `logicalDisks` is nil and no error, I think it should return also.

Reply:
Yeah, sorry, I've misread this line